inspirehep/modules/workflows/views/callback.py

```diff
@@ -153,12 +179,79 @@ def webcoll_callback():
         )
         db.session.delete(pending_record)
         db.session.commit()
-        if response[recid].get('status') == 'failed':
-            continue
 
     return jsonify(response)
 
 
+def _robotupload_has_error(result):
+    recid = int(result.get('recid'))
+    if not result.get('success'):
+        message = result.get(
+            'error_message',
+            'No error message from robotupload.'
+        )
+    elif recid < 0:
+        message = result.get(
+            'error_message',
+            'Failed to create record on robotupload.',
+        )
+    else:
+        return False, ''
+
+    return True, message
+
+
+def _parse_robotupload_result(result, workflow_id):
+    response = {}
+    recid = int(result.get('recid'))
+
+    already_pending_ones = WorkflowsPendingRecord.query.filter_by(
+        record_id=recid,
+    ).all()
+    if already_pending_ones:
+        current_app.logger.warning(
+            'The record %s was already found on the pending list.',
+            recid
+        )
+        response = {
+            'status': 'failed',
+            'message': 'Recid %s already in pending list.' % recid,
+        }
+        return response
```

Comment:
This is triggered whenever we get a robotupload for a record that is already pending to get the webcall, I don't think we should ignore it as that possibly means that the first try got messed up somewhere and the second failed to detect that it was already there, maybe we should define more in detail all the possible cases.